### ISSUE TITLE
Add pseudo-spectral eigenvalue option and staggered grid documentation

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#40]
## Summary

Add a **pseudo-spectral (PS) eigenvalue** option to the Layer 0 functional solvers (currently FD2-only), and write comprehensive **documentation** covering the staggered grid framework, the full BC × grid × transform mapping, and usage examples.

**Blocked by:** #38 (eigenvalue functions and solvers must exist first)

### Motivation

The current Layer 0 solvers (`solve_helmholtz_dst`, `solve_helmholtz_dct`, etc.) use **finite-difference (FD2)** eigenvalues — they are exact inverses of the 5-point stencil Laplacian. This is correct for finite-difference/finite-volume codes.

The Layer 1 `SpectralHelmholtzSolver{1,2,3}D` classes use **continuous/pseudo-spectral (PS)** eigenvalues via the `FourierGrid` wavenumbers. This is correct for pseudo-spectral methods.

Currently there's no way to use PS eigenvalues with the DST/DCT-based solvers (Dirichlet/Neumann). Both SailFFish and PoisFFT support both eigenvalue types for all BC/grid combinations.

### PS vs FD2 Eigenvalue Comparison

From SailFFish Table 1 and PoisFFT Table 3:

| BC | Grid | PS Eigenvalue | FD2 Eigenvalue |
|---|---|---|---|
| Dirichlet | Regular | `−(π(k+1)/L)²` | `−(2sin(π(k+1)/(2(N+1)))/dx)²` |
| Dirichlet | Staggered | `−(π(k+1)/L)²` | `−(2sin(π(k+1)/(2N))/dx)²` |
| Neumann | Regular | `−(πk/L)²` | `−(2sin(πk/(2(N−1)))/dx)²` |
| Neumann | Staggered | `−(πk/L)²` | `−(2sin(πk/(2N))/dx)²` |
| Periodic | Either | `−(2πk/L)²` (cyclic) | `−(2sin(πk/N)/dx)²` |

**Key difference:** PS eigenvalues use the domain length L in the denominator; FD2 eigenvalues use the grid spacing dx inside a `sin²`. They agree for low wavenumbers (`k ≪ N`) but diverge near Nyquist. PS gives spectral accuracy for smooth solutions; FD2 gives O(h²) convergence but is the correct inverse for finite-difference discretizations.

#### A. New PS Eigenvalue Functions (`eigenvalues.py`)

```python
def dst1_eigenvalues_ps(N: int, L: float) -> Float[Array, " N"]:
    """PS eigenvalues for DST-I (Dirichlet, regular grid).
    λ_k = −(π(k+1)/L)²,   k = 0, …, N−1
    """
    k = jnp.arange(N)
    return -(jnp.pi * (k + 1) / L) ** 2

def dst2_eigenvalues_ps(N: int, L: float) -> Float[Array, " N"]:
    """PS eigenvalues for DST-II (Dirichlet, staggered grid).
    Same formula as DST-I PS: λ_k = −(π(k+1)/L)²
    """
    k = jnp.arange(N)
    return -(jnp.pi * (k + 1) / L) ** 2

def dct1_eigenvalues_ps(N: int, L: float) -> Float[Array, " N"]:
    """PS eigenvalues for DCT-I (Neumann, regular grid).
    λ_k = −(πk/L)²,   k = 0, …, N−1.  λ_0 = 0.
    """
    k = jnp.arange(N)
    return -(jnp.pi * k / L) ** 2

def dct2_eigenvalues_ps(N: int, L: float) -> Float[Array, " N"]:
    """PS eigenvalues for DCT-II (Neumann, staggered grid).
    Same formula as DCT-I PS: λ_k = −(πk/L)²
    """
    k = jnp.arange(N)
    return -(jnp.pi * k / L) ** 2

def fft_eigenvalues_ps(N: int, L: float) -> Float[Array, " N"]:
    """PS eigenvalues for FFT (periodic). Cyclic wavenumbers.
    λ_k = −(2πk/L)² for k < N/2, −(2π(N−k)/L)² for k ≥ N/2
    """
    k = jnp.arange(N)
    k_phys = jnp.where(k < N // 2, k, N - k)
    return -(2 * jnp.pi * k_phys / L) ** 2
```

Note: PS eigenvalues take domain length `L` as parameter (not grid spacing `dx`), since `L` is the natural scale for spectral methods. This differs from FD2 eigenvalues which take `dx`.

#### B. Add `approximation` Parameter to Layer 0 Solvers

Add an `approximation` parameter to all functional solvers:

```python
def solve_helmholtz_dst(
    rhs, dx, dy, lambda_=0.0,
    approximation: Literal["fd2", "spectral"] = "fd2",
):
```

When `approximation="spectral"`, compute `L = N * dx` (or appropriate formula per BC/grid type) and use PS eigenvalues. When `"fd2"` (default), use existing FD2 eigenvalues. Default remains `"fd2"` for backwards compatibility.

**Grid spacing → domain length conversion** depends on BC/grid type (from PoisFFT `grid_dx`):

| BC | Grid | L from dx and N |
|---|---|---|
| Periodic | Either | `L = N × dx` |
| Dirichlet | Regular | `L = (N+1) × dx` |
| Dirichlet | Staggered | `L = N × dx` |
| Neumann | Regular | `L = (N−1) × dx` |
| Neumann | Staggered | `L = N × dx` |

#### C. Tests

- **Spectral convergence test:** With PS eigenvalues and a smooth test function, error should drop to machine precision as N increases. Use the test functions from SailFFish Section 4.1:
  - Dirichlet: `f(x) = sin(4π x/L)`
  - Neumann: `f(x) = cos(4π x/L)`

- **O(h²) convergence test:** With FD2 eigenvalues, error should decrease as O(N⁻²). Plot error vs N on a log-log scale and check slope ≈ −2.

- **Cross-validation:** For the same input, PS and FD2 solutions should agree for low wavenumbers but differ for high wavenumbers.

### A. New Theory Page: `docs/theory/staggered_grids.md`

Comprehensive documentation covering:

1. **Regular vs Staggered Grid Definitions**
   - Regular (vertex-centered): grid points on cell boundaries, including domain boundary
   - Staggered (cell-centered): grid points at cell centers, half-cell offset from boundary
   - Diagrams showing point positions for both types
   - Grid position formulas:
     ```
     Regular:   x_i = x_l + i·dx,           i = 0, …, N_s
     Staggered: x_i = x_l + (i + 0.5)·dx,   i = 0, …, N_c − 1
     ```

2. **The Full BC × Grid → Transform Mapping Table**
   - Reproduce the unified table from the plan (SailFFish Table 1 + PoisFFT Tables 2–4)
   - Explain why each BC/grid combination requires a specific transform type
   - Show that all 4 DST and 4 DCT types have a use case

3. **FD2 vs PS Eigenvalue Comparison**
   - Formulas for both types, side by side
   - When they agree (low wavenumbers) and when they diverge (near Nyquist)
   - Which to use: FD2 for finite-difference codes, PS for spectral codes
   - Reference diagram showing eigenvalue curves (similar to existing `eigenvalues_comparison.png`)

4. **Grid Spacing Conventions**
   - How dx relates to L and N for each BC/grid type
   - Number of transform points vs number of solution points

5. **Physical Motivation**
   - Why staggered grids matter for CFD (pressure correction, Arakawa C-grid)
   - Example: incompressible Navier-Stokes projection method

6. **References**
   - Saverin (2023), SailFFish paper
   - Fuka (2014), PoisFFT paper
   - Schumann & Sweet (1988)
   - Wiegmann (1999), "Fast Poisson, fast Helmholtz and fast linear elastostatic solvers"

### B. Update Existing Docs

**`docs/theory/spectral_transforms.md`:**
- Add cross-reference to new staggered grids page
- In Section 5 (Finite-Difference Eigenvalues), note that the current formulas are for specific BC/grid combinations and link to the complete table

**`docs/elliptic_solvers_guide.md`:**
- Add examples showing the new solvers:
  ```python
  from spectraldiffx import solve_helmholtz_dst2, solve_helmholtz_dct1

  # Staggered Dirichlet (cell-centered, e.g., pressure in channel flow)
  psi = solve_helmholtz_dst2(rhs, dx, dy)

  # Regular Neumann (vertex-centered)
  psi = solve_helmholtz_dct1(rhs, dx, dy)

  # Mixed BCs (periodic-x, neumann_stag-y)
  psi = solve_helmholtz_2d(rhs, dx, dy, bc_x="periodic", bc_y="neumann_stag")
  ```
- Add a section on choosing the right solver for your problem

### C. API Reference

Auto-generated API docs for all new public functions. Ensure all docstrings follow NumPy style with:
- Mathematical formula
- Parameters with types
- Returns with shape annotations
- Notes on which BC/grid type the function corresponds to

## References

- SailFFish paper Sections 2.4, 4.1 (eigenvalues, convergence validation)
- PoisFFT paper Section 2 (solution methods, eigenvalue tables)
- Full plan: `spectraldiffx/.plans/staggered-grid-solvers.md` (Phases 5 + 6)
